### PR TITLE
Datadir support  #184

### PR DESCRIPTION
**Pull request by naokiri**
This change adds support for installing read-only architecture-independent data files.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -127,6 +127,19 @@ install_subdir = "gstreamer-1.0"
 versioning = false
 ```
 
+### Shared Data Install
+```toml
+[package.metadata.capi.shared.data]
+# Can be used to suppress installing shared data files. Defaults to false.
+enabled = false
+# Relative path from the project root where the shared data file is. 
+# Defaults to "shared/data".
```
Comment (naokiri) [inline README.md:136]: I am very hesitant on using this default. Is there any good well-known directory structure for this kind of file?
At first I wanted to use  `build.rs` outputs so I tried to read `OUT_DIR` ( https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-build-scripts ) but it wasn't available on running `cargo cinstall`

**Review by lu-zero — CHANGES_REQUESTED** [review]
I'd rather have this discussion with upstream cargo if possible. If nothing happens on their side we can look into it.

File: README.md (hunk 1 above)
Comment (naokiri) [reply README.md:136]: This is the origin of the data to be installed, so the destination doesn't matter. 
In this example, `src/data` is copied to $datadir/$install_dir, which is `/usr/local/share/libfoo`

**Comment by lu-zero** [comment]
All the machinery to support this use case is in the master, if you have time we can add the remaining bits this month :)

**Pull request closed without merging**